x-pack/plugin/rank-text-similarity/src/main/java/org/elasticsearch/xpack/rank/textsimilarity/TextSimilarityRankRetrieverBuilder.java

```diff
@@ -0,0 +1,142 @@
+import org.elasticsearch.search.retriever.RetrieverBuilder;
+import org.elasticsearch.search.retriever.RetrieverParserContext;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xpack.core.XPackPlugin;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.search.rank.RankBuilder.DEFAULT_RANK_WINDOW_SIZE;
+import static org.elasticsearch.xpack.rank.textsimilarity.TextSimilarityRankPlugin.NAME;
+
+public class TextSimilarityRankRetrieverBuilder extends RetrieverBuilder {
+    public static final NodeFeature NODE_FEATURE = new NodeFeature("text_similarity_rank_retriever_supported");
+
+    public static final ParseField RETRIEVER_FIELD = new ParseField("retriever");
+    public static final ParseField FIELD_FIELD = new ParseField("field");
+    public static final ParseField RANK_WINDOW_SIZE_FIELD = new ParseField("rank_window_size");
+    public static final ParseField INFERENCE_ID_FIELD = new ParseField("inference_id");
+    public static final ParseField INFERENCE_TEXT_FIELD = new ParseField("inference_text");
+    public static final ParseField MIN_SCORE_FIELD = new ParseField("min_score");
+
+    public static final ObjectParser<TextSimilarityRankRetrieverBuilder, RetrieverParserContext> PARSER = new ObjectParser<>(
+        NAME,
+        TextSimilarityRankRetrieverBuilder::new
+    );
+
+    static {
+        PARSER.declareObject((r, v) -> r.retrieverBuilder = v, (p, c) -> {
+            RetrieverBuilder retrieverBuilder = RetrieverBuilder.parseInnerRetrieverBuilder(p, c);
+            c.trackSectionUsage(NAME + ":" + RETRIEVER_FIELD.getPreferredName());
+            return retrieverBuilder;
+        }, RETRIEVER_FIELD);
+
+        PARSER.declareField((r, v) -> r.field = v, (p, c) -> {
+            String field = p.text();
+            c.trackSectionUsage(NAME + ":" + FIELD_FIELD.getPreferredName());
+            return field;
+        }, FIELD_FIELD, ObjectParser.ValueType.STRING);
+
+        PARSER.declareField((r, v) -> r.rankWindowSize = v, (p, c) -> {
+            int rankWindowSize = p.intValue();
+            c.trackSectionUsage(NAME + ":" + RANK_WINDOW_SIZE_FIELD.getPreferredName());
+            return rankWindowSize;
+        }, RANK_WINDOW_SIZE_FIELD, ObjectParser.ValueType.INT_OR_NULL);
+
+        PARSER.declareField((r, v) -> r.inferenceId = v, (p, c) -> {
+            String inferenceId = p.text();
+            c.trackSectionUsage(NAME + ":" + INFERENCE_ID_FIELD.getPreferredName());
+            return inferenceId;
+        }, INFERENCE_ID_FIELD, ObjectParser.ValueType.STRING);
+
+        PARSER.declareField((r, v) -> r.inferenceText = v, (p, c) -> {
+            String inferenceText = p.text();
+            c.trackSectionUsage(NAME + ":" + INFERENCE_TEXT_FIELD.getPreferredName());
```

Comment:
Why are we tracking every individual section?

Reply:
Again, I was following a pattern I've seen in `StandardRetrieverBuilder`, where [every section is being tracked](https://github.com/elastic/elasticsearch/blob/b0d9f95e534e07aca216a556bdaa5e137fe87bbd/server/src/main/java/org/elasticsearch/search/retriever/StandardRetrieverBuilder.java#L56).

Telemetry requirements are going to be refined after 8.15. Until then I can remove the tracking of those fields that are mandatory in this retriever as it's redundant.

Reply:
Removed tracking of sections.